src/piadm.sh

```diff
@@ -655,13 +675,250 @@ ispoolenabled() {
+the word "ipxe" in platform/etc/version/platform, are here so there's
+something in the platform/ directory to prevent piadm (especially
+older versions) from thinking something is wrong.
+EOF
+}
+
+# Given a bootstamp, install a Platform Image as a backup for the compute node.
+# PWD is /${bootfs} at this point.
+install_pi_CN() {
+	vecho "Installing as a backup Platform image PI stamp $1"
+
+	if [[ -d ./platform-$1 ]]; then
+		vecho "PI stamp $1 already installed."
+		installstamp=$1
+		return 0
+	fi
+
+	# Obtain at least unix and boot archive.
+
+	# For now, use bootparams to get the URL needed, and pull
+	# files from there.  If there's a better way to obtain things, use it.
+	unix_path=$(bootparams | grep boot-file | awk -F= '{print $2}')
+	if [[ "$1" == "$CNAPI_DEFAULT_PI" ]]; then
+		# We need to edit out the bootstamp part.  Count on path
+		# having "os/STAMP/" in it.
+		unix_path=$(echo "$unix_path" | sed "s/os\/[0-9TZ]*\//os\/$CNAPI_DEFAULT_PI\//g")
+	fi
+	archive_prefix=$(echo "$unix_path" | sed 's/kernel\/amd64\/unix/amd64/g')
+
+	# Reality check the buildstamp passed, which will become installstamp,
+	# is in the unix_path.
+	echo "$unix_path" | grep -q "$1" || return 1
+
+	installstamp=$1
+	vecho "making platform-$installstamp directories"
+	mkdir -p platform-"$installstamp"/etc/version
+	echo "$installstamp" > platform-"$installstamp"/etc/version/platform
+	mkdir -p platform-"$installstamp"/i86pc/kernel/amd64
+	mkdir -p platform-"$installstamp"/i86pc/amd64
+	# To enable a platform/ component in the boot file pathname to confirm
+	# "unix" is also in the boot archive (as /platform/..../unix).
+	ln -s . platform-"$installstamp"/platform
+
+	vecho "Pulling unix"
+	"${CURL[@]}" "$unix_path" > \
+		platform-"$installstamp"/i86pc/kernel/amd64/unix || return 1
+	for file in boot_archive boot_archive.hash boot_archive.manifest \
+		boot_archive.gitstatus; do
+		vecho "Pulling" "$file"
+		"${CURL[@]}" "${archive_prefix}"/"${file}" > \
+			platform-"$installstamp"/i86pc/amd64/"${file}" || \
+			return 1
+	done
+
+	return 0
+}
+
+# Enabling a bootable pool, specifically for a Triton Compute Node.
+bringup_CN() {
+	# Bootfs is already set at this point.
+
+	if [[ "$CNAPI_DEFAULT_PI" != "$activestamp" ]]; then
+		vecho "Current booted PI $activestamp is not default PI" \
+			"$CNAPI_DEFAULT_PI"
+	fi
+
+	# First install ipxe in $bootfs.
+	cd "/${bootfs}" || fatal "Could not chdir to /$bootfs"
+	# Clobber everything in $bootfs.  We do not care about dot-files.
+	rm -rf ./*
+
+	# The "platform-ipxe" directory for on-disk iPXE is a placeholder.
+	# We put a README (see cat_readme() above) and the string "ipxe"
+	# for the PI-stamp.
+	mkdir -p ./platform-ipxe/etc/version
+	cat_readme > ./platform-ipxe/README
+	echo "ipxe" > ./platform-ipxe/etc/version/platform
+
+	# Now we set up the "etc" directory, which contains versions of
+	# both loader ("boot") and iPXE ("ipxe").
+	mkdir -p etc/version
+	cp -f ${TRITON_IPXE_ETC}/version/* etc/version/.
```

Comment:
You should quote this variable for safety....example:
```
link@skullcanyon:~$ foo="foo bar"
link@skullcanyon:~$ ./args.sh ${foo}
2 args: <foo> <bar>
link@skullcanyon:~$ ./args.sh "${foo}"
1 args: <foo bar>
```

Reply:
These are hardcoded and not subject to environment or user-input weirdness.